client/js/widgets/button.js

```diff
@@ -44,22 +44,186 @@ export class Button extends Widget {
   applyVariables(field, variables, problems) {
     if(Array.isArray(field.applyVariables)) {
       for(const v of field.applyVariables) {
-        if(v.parameter && v.variable) {
-          field[v.parameter] = (v.index === undefined) ? variables[v.variable] : variables[v.variable][v.index];
+        if (v.var_operation)
+          v.operation = variables[v.var_operation];
+        if (v.operation) {
+          let op = [null, 1, 1, 1];
+          Object.entries(v).forEach(([key, value]) => {
+            op[key[key.length-1]] = value;
+            if (['var_operand1', 'var_operand2', 'var_operand3'].indexOf(key) >= 0)
+              op[key[key.length-1]] = variables[value].index !== undefined ? variables[value][value.index] : variables[value];
```

Comment:
This line looks very broken to me. You check if the given variable has an `index` and then use an `index` of the `value`. Is it even planned to use an index here? Seems too complicated.

Reply:
This wasn't added by me. I remember this was added a while back, but I cannot remember the exact use case.

Reply:
Oh wait, I think I see where I messed up.

Reply:
@ArnoldSmith86 With introducing compute operations in applyVariables, we actually no longer need the index parameter, as we could simply use getIndex as a prior compute step. I wonder if that's safe to remove the index parameter as it's never been documented and quite unlikely that any game uses it?

Reply:
I'll check if it's being used anywhere on the server currently. Might be save.

Reply:
> With introducing compute operations in applyVariables, we actually no longer need the index parameter, as we could simply use getIndex as a prior compute step. I wonder if that's safe to remove the index parameter as it's never been documented and quite unlikely that any game uses it?

I think that the original impetus for adding this was a request from me regarding a need I had in Dots. I worked around it and am happy with the result, so I at least never ended up using it.